controller-server/src/test/java/com/yahoo/vespa/hosted/controller/application/EndpointTest.java

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.application;
+
+import com.yahoo.config.provision.ApplicationId;
+import com.yahoo.config.provision.ClusterSpec;
+import com.yahoo.config.provision.RotationName;
+import com.yahoo.config.provision.SystemName;
+import com.yahoo.config.provision.zone.ZoneId;
+import com.yahoo.vespa.hosted.controller.application.Endpoint.Port;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author mpolden
+ */
+public class EndpointTest {
+
+    private static final ApplicationId app1 = ApplicationId.from("t1", "a1", "default");
+    private static final ApplicationId app2 = ApplicationId.from("t2", "a2", "i2");
+
+    @Test
+    public void test_global_endpoints() {
+        RotationName rotation = RotationName.from("default"); // Always default for non-direct routing
+
+        Map<String, Endpoint> tests = Map.of(
+                // Legacy endpoint
+                "http://a1.t1.global.vespa.yahooapis.com:4080/",
+                Endpoint.of(app1).target(rotation).on(Port.plain(4080)).legacy().in(SystemName.main),
+
+                // Legacy endpoint with TLS
+                "https://a1--t1.global.vespa.yahooapis.com:4443/",
+                Endpoint.of(app1).target(rotation).on(Port.tls(4443)).legacy().in(SystemName.main),
+
+                // Main endpoint
+                "https://a1--t1.global.vespa.oath.cloud:4443/",
+                Endpoint.of(app1).target(rotation).on(Port.tls(4443)).in(SystemName.main),
+
+                // Main endpoint in CD
+                "https://cd--a1--t1.global.vespa.oath.cloud:4443/",
+                Endpoint.of(app1).target(rotation).on(Port.tls(4443)).in(SystemName.cd),
+
+                // Main endpoint with direct routing and default TLS port
+                "https://a1.t1.global.vespa.oath.cloud/",
+                Endpoint.of(app1).target(rotation).on(Port.tls()).directRouting().in(SystemName.main),
+
+                // Main endpoint with custom rotation name
+                "https://r1.a1.t1.global.vespa.oath.cloud/",
+                Endpoint.of(app1).target(RotationName.from("r1")).on(Port.tls()).directRouting().in(SystemName.main),
+
+                // Main endpoint for custom instance
+                "https://i2.a2.t2.global.vespa.oath.cloud/",
+                Endpoint.of(app2).target(rotation).on(Port.tls()).directRouting().in(SystemName.main),
+
+                // Main endpoint for custom instance with custom rotation name
+                "https://r2.i2.a2.t2.global.vespa.oath.cloud/",
+                Endpoint.of(app2).target(RotationName.from("r2")).on(Port.tls()).directRouting().in(SystemName.main),
+
+                // Main endpoint in public system
+                "https://a1.t1.global.public.vespa.oath.cloud/",
```

Comment:
I'm not sure we should have "global" here, at least we did not include this when it was discussed. Probably my fault as I see I used this in examples on slack.

Reply:
Discussed offline: Keep "global" for now and revisit when we change to the final suffix.